src/nccl_ofi_net.c

```diff
@@ -87,6 +87,66 @@ float net_latency = .0;
 /* Size of a memory page */
 long system_page_size = -1;
 
+/*
+ * @brief	Allocate memory region for memory registration
+ *
+ * This function allocates memory that covers full page aligned.
+ *
+ * Internally allocated memory that is registered is required to cover
+ * full memory pages. For more information, see functions
+ * `register_internal_mr_buffers()` and `reg_internal_mr_ep()`.
+ *
+ * To free deallocate the memory region, function
+ * nccl_net_ofi_dealloc_mr_buffer() must be used.
+ *
+ * @param	size
+ *		Size of the memory region. Must be a multiple of system memory page size.
+ * @return	Pointer to memory region. Memory region is aligned to system memory page size.
+ * @return	0, on success
+ *		error, on others
+ */
+int nccl_net_ofi_alloc_mr_buffer(size_t size, void **ptr)
+{
+	assert(system_page_size > 0);
+	assert(NCCL_OFI_IS_ALIGNED(size, system_page_size));
```

Comment:
We don't need to enforce that `size` is a multiple of page-size because `mmap()` guarantees to allocate on page size resolution. It's sufficient that we use `mmap()` for allocating a MR buffer.